desc/magnetic_fields.py

```diff
@@ -170,21 +170,21 @@ def __sub__(self, x):
         return self.__add__(-x)
 
     @abstractmethod
-    def compute_magnetic_field(self, coords, params=None, basis="rpz", grid=None):
+    def compute_magnetic_field(self, grid, params=None, basis="rpz", source_grid=None):
```

Comment:
I thought our old notation of `coords` vs `grid` was confusing, so I updated it for every `compute_magnetic_field` function. "coords" sounds like coordinate system, but that is what the `basis` argument is for. Now `grid` is the grid where the magnetic field is being evaluated at, and `source_grid` is the optional grid for discretizing the current source. Both can be either `Grid` types or an array of the grid nodes.

Reply:
I think source grid always be either Grid or None, not an array, since some magnetic field classes need the other info from the grid class.

Also if we want to rename things I'd prefer eval_grid and source_grid to make them both clear.

Reply:
^ yea I think the `FourierCurrentPotentialField` has to have a `Grid` object since it computes `K` using the `surface.compute` method